### ISSUE TITLE
Support for ENSEMBL API and/or ENSEMBL REST API

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#512]
We have ``Bio.Entrez`` for the NCBI Entrez Utilities, ``Bio.TogoWS`` for the Togo Web Services in Japan, etc. It would be great to have something similar for ENSEMBL.

See http://www.ensembl.org/info/docs/api/index.html and/or https://github.com/Ensembl/ensembl-rest/wiki including https://github.com/Ensembl/ensembl-rest/wiki/Example-Python-Client

**Comment by biomadeira** [comment]
@peterjc I have some working tools to interface this api (for getting variation data). They absolutely need more work before adding them here but perhaps if someone is already working on this I would be totally happy to team up and help... Cheers

**Comment by chiragmatkar** [comment]
Any plans for for collaborating on this project.I would like to help with this as well

**Comment by peterjc** [comment]
I've not done anything on this myself. Over to @biomadeira - do you have something on GitHub we (e.g. @chiragmatkar ) could look at?

**Comment by chiragmatkar** [comment]
Unfortunately No, Peter.i was working as Perl Developer in Bioinformatics few years back and playing with the EnsembL APi for my projects.Currently i am working with python , hence i thought could give a shot if anyone has starting working on it .

**Comment by chiragmatkar** [comment]
Yup i would like to work with @biomadeira  if something is up with him.

**Comment by biomadeira** [comment]
I am currently putting some things together with @tbrittoborges (https://github.com/tbrittoborges) and we will upload it soon. Very much looking into joining efforts then with @chiragmatkar on this. Back to you soon! 
Cheers

**Comment by tbrittoborges** [comment]
So, @peterjc, what are the objectives of the module? Just fetching sequences from Ensembl REST API, for now?

**Comment by chiragmatkar** [comment]
Yes i think we could start with basic.Slowly we can build few things from
below:

   - Class and package names are mixed-case words that begin with capital
   letters.

   Bio::EnsEMBL::Gene, Bio::EnsEMBL::Exon, Bio::EnsEMBL::Slice,
   Bio::EnsEMBL::DBSQL::GeneAdaptor



On Wed, Jun 10, 2015 at 3:19 PM, Thiago Britto Borges <
notifications@github.com> wrote:

> So, @peterjc <https://github.com/peterjc>, what are the objectives of the
> module? Just fetching sequences from Ensembl REST API, for now?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biopython/biopython/issues/512#issuecomment-110675144>
> .
>



-- 

Thanks and Regards,
Chirag Matkar

**Comment by chiragmatkar** [comment]
There is a wrapper written for interfacing ENSEMBLE REST
https://github.com/pyOpenSci/pyEnsemblRest

On Wed, Jun 10, 2015 at 5:28 PM, chirag matkar <chirag.matkar@gmail.com>
wrote:

> Yes i think we could start with basic.Slowly we can build few things from
> below:
>
>    - Class and package names are mixed-case words that begin with capital
>    letters.
>
>    Bio::EnsEMBL::Gene, Bio::EnsEMBL::Exon, Bio::EnsEMBL::Slice,
>    Bio::EnsEMBL::DBSQL::GeneAdaptor
>
>
>
> On Wed, Jun 10, 2015 at 3:19 PM, Thiago Britto Borges <
> notifications@github.com> wrote:
>
>> So, @peterjc <https://github.com/peterjc>, what are the objectives of
>> the module? Just fetching sequences from Ensembl REST API, for now?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/biopython/biopython/issues/512#issuecomment-110675144>
>> .
>>
>
>
>
> --
>
> Thanks and Regards,
> Chirag Matkar
>
>
>



-- 

Thanks and Regards,
Chirag Matkar

**Comment by peterjc** [comment]
Note pyEnsemblRest is under the GPL v3, so you can't simply copy it or bits of it for Biopython. We could ask the authors if they would be interested in contributing anything to Biopython (with a re-licensing of the code).
